py_ocpi/credentials/v_2_2_1/api/cpo.py

```diff
@@ -1,10 +1,113 @@
+
+        # Retrieve the versions and endpoints from the client
+        async with httpx.AsyncClient() as client:
+            authorization_token = f'Token {credentials_client_token}'
+            response_versions = await client.get(credentials.url,
+                                                 headers={'authorization': authorization_token})
+            versions = response_versions.json()['data'][0]
+            response_endpoints = await client.get(versions['url'],
+                                                  headers={'authorization': authorization_token})
+
+        if response_endpoints.status_code == fastapistatus.HTTP_200_OK:
+            # Store client credentials
+            endpoints = response_endpoints.json()['data'][0]
+            await crud.create(ModuleID.CredentialsAndRegistrations, ServerCredentials(
+                cred_token_b=credentials.token,
+                versions=versions,
+                endpoints=endpoints
+            ))
+
+            # Generate new credentials for sender
+            new_credentials = await crud.create(ModuleID.CredentialsAndRegistrations, Credentials(url=versions['url']))
+            return OCPIResponse(
+                data=[adapter.credentials_adapter(new_credentials).dict()],
+                **status.OCPI_1000_GENERIC_SUCESS_CODE
+            )
+    except ValidationError:
+        return OCPIResponse(
+            data=[],
+            **status.OCPI_3001_UNABLE_TO_USE_CLIENTS_API,
+        )
+
+
+@router.put("/", response_model=OCPIResponse)
+async def update_credentials(credentials: Credentials, crud=Depends(get_crud), adapter=Depends(get_adapter)):
+    try:
+        # Check if the client is already registered
+        credentials_client_token = credentials.token
+        server_cred = await crud.get(ModuleID.CredentialsAndRegistrations, credentials_client_token)
+        if not server_cred:
+            raise HTTPException(fastapistatus.HTTP_405_METHOD_NOT_ALLOWED, "Client is not registered")
+
+        # Retrieve the versions and endpoints from the client
+        async with httpx.AsyncClient() as client:
+            authorization_token = f'Token {credentials_client_token}'
+            response_versions = await client.get(credentials.url, headers={'authorization': authorization_token})
+            versions = response_versions.json()['data'][0]
+            response_endpoints = await client.get(versions['url'], headers={'authorization': authorization_token})
+
+        if response_endpoints.status_code == fastapistatus.HTTP_200_OK:
+            # Update server credentials to access client's system
+            endpoints = response_endpoints.json()['data'][0]
+            await crud.update(ModuleID.CredentialsAndRegistrations,
+                              {'versions': versions,
+                               'endpoints': endpoints,
+                               'cred_token_b': credentials_client_token},
```

Comment:
if you're using ServerCredentials then please use it in all places or otherwise if you want to use the dict use dict for all. check Line 58